feature-libs/checkout/scheduled-replenishment/components/services/checkout-replenishment-form-service.spec.ts

```diff
@@ -0,0 +1,72 @@
+import { TestBed, waitForAsync } from '@angular/core/testing';
+import { ScheduleReplenishmentForm } from '@spartacus/core';
+import { CheckoutReplenishmentFormService } from './checkout-replenishment-form-service';
+
+const newReplenishmentFormData: ScheduleReplenishmentForm = {
+  recurrencePeriod: 'test-period',
+  numberOfWeeks: 'test-num-weeks',
+};
+
+describe('Checkout Replenishment Form Service', () => {
+  let service: CheckoutReplenishmentFormService;
+
+  beforeEach(
+    waitForAsync(() => {
+      TestBed.configureTestingModule({
+        providers: [CheckoutReplenishmentFormService],
+      });
+    })
+  );
+
+  beforeEach(() => {
+    service = TestBed.inject(CheckoutReplenishmentFormService);
+  });
+
+  it('should get default schedule replenishment form data', () => {
+    const defaultFormData: ScheduleReplenishmentForm = service.defaultFormData;
+    defaultFormData.nthDayOfMonth = '2';
+    let expectedFormData: ScheduleReplenishmentForm | undefined;
```

Comment:
I don't think we need to add the | undefined type as the form will never be undefined

Reply:
it's added because TS complains the variable is used before it is initialized